### PR TITLE
Enhancement: QR Code UI update and customizations

### LINKED ISSUE #171: GSSoC: QR Code page UI and Customizations
### Description

I would like to code the UI for QR code page and add customizations to the generate qr code feature.

### Changes You need to make

Will add feature to change:
1. color
2. size
3. margin
4. format
etc.. for the qr code to be generated ....

Files to be changed: 
- client\src\components\pages\qr-codes.tsx
- building new components

I have gone through the qr code api to check for parameters that I can add to the qr code...

Checkout the UI design I have built on figma for the QR code page... any further improvements are open for changes...

### Screenshots

image

### Checklist

Before submitting your pull request, ensure that you have completed the following tasks:

- [x] Gone through the [rules and guidelines](https://github.com/oxiton-foundation/click-metrics/issues/24) of the project.
- [x] When you are creating PR, your PR should title follows the specified format: [Feature/Enhancement/Bug Fix]: Brief description of the change.
- [x] Attached OSSI or GSSoC in the heading of the Issue.
- [x] I have checked this issue is not already covered in another issue.

### Expected Time to Complete

around 2-4 days

### Additional Notes

I will make the page mobile responsive as well and add most api features to customize the QR Code based on user needs....
I will also add a loading animation on qr code image when generate qr is clicked to provide a feedback to the user.

If you like the changes, please assign the Issue to me and add gssoc labels and appropriate level....

### PR DESCRIPTION
**Pull request by anisharma07**
### Related Issue(s)

- Fixes #171 

### Description

Now we can edit the color, background color, resolution(size), margin, format of the qr code....

### Changes Made

Provide a detailed list of the changes made in this pull request. Include specific files, components, or functionalities affected by your modifications.

- client\src\components\pages\qr-codes.tsx

### Screenshots (if applicable)

Attach screenshots or Videos demonstrating the visual changes introduced by your pull request. This helps reviewers understand the user interface modifications more effectively.

https://github.com/oxiton-foundation/click-metrics/assets/126901961/e9aceb34-3de4-4d46-aea1-928619c99629

### Checklist

Before submitting your pull request, ensure that you have completed the following tasks:

- [x] Code follows the project's style and conventions.
- [x] Code has been tested thoroughly, and all tests pass successfully.
- [x] Documentation has been updated to reflect any relevant changes.
- [x] All new and modified code is adequately commented for clarity.
- [x] Pull request title follows the specified format: [Feature/Enhancement/Bug Fix]: Brief description of the change.

### CODE REVIEW

**Comment by Smitkhobragade** [comment]
@anisharma07 . Instead od downloading SVG. can we make it png or jpg. which is ewasy to be shared.
or let the user have option which format to download. like we have a button in github similar to that?

I am pasting one example

https://github.com/oxiton-foundation/click-metrics/assets/100017129/138ebd8d-5c6f-48fb-9e52-def69fdb1faf

Like here it is having 
1. Create new file
2. Upload File

we can have options like
1. Download  PNG
2. Download SVG
...

Like that

**Comment by anisharma07** [comment]
> @anisharma07 . Instead od downloading SVG. can we make it png or jpg. which is ewasy to be shared. or let the user have option which format to download. like we have a button in github similar to that?
> 
> I am pasting one example
> 
>  Recording.2024-06-14.114537.mp4 
> Like here it is having
> 
> 1. Create new file
> 2. Upload File
> 
> we can have options like
> 
> 1. Download  PNG
> 2. Download SVG
>    ...
> 
> Like that

Oh yeah! I have given that option .. user can choose the format in which they want to download the qr code ..... whichever is format is selected below will be format that will be downloaded

image

**Comment by Swarnendu0123** [comment]
I guess this will be a great feature, if we can give options to the user to choose the format of the image. You can implement it, but create a separate PR, which will be easy to check. It will be level 2. 

Any suggestions @Smitkhobragade ?

**Comment by Smitkhobragade** [comment]
Umm. Nothing much. But It will be a nice feature for a user to have specific options...

**Comment by anisharma07** [comment]
@Smitkhobragade @Swarnendu0123    
Can you give me level 3 for this PR.... I will figure out how to make QR download in preffered format , without needing to choose format before generating it ....  similar to this qr generator website
image

**Comment by Swarnendu0123** [comment]
Cool, so do you feel this PR is ready for the evaluation?

**Comment by anisharma07** [comment]
@Smitkhobragade @Swarnendu0123 

Now, you can merge the PR , I have included choice for user to download the qr in their preferred format 👇
https://github.com/oxiton-foundation/click-metrics/assets/126901961/e9aceb34-3de4-4d46-aea1-928619c99629

File: client/src/components/pages/qr-codes.tsx [hunk 1]
```diff
@@ -1,21 +1,64 @@
 import React, { useState } from "react";
 import { message } from "antd";
 import defaultQr from "../../assets/qrcode.png";
+// import { set } from "react-hook-form";
```
Comment (Swarnendu0123) [inline client/src/components/pages/qr-codes.tsx:4]: Remove the comments

File: client/src/components/pages/qr-codes.tsx [hunk 2]
```diff
@@ -1,21 +1,64 @@
...
 
 const Qrcode: React.FC = () => {
-  const [img, setImg] = useState<string>(defaultQr); // Initialize with default QR image path
-  const [qrinput, setQrinput] = useState<string>("click_matrics");
+  type ImageFormats = {
```
Comment (Swarnendu0123) [inline client/src/components/pages/qr-codes.tsx:7]: Move the types to types folder and rename the type variables such that suffix should include the word `type`. Eg:  ImageFormatType.

**Comment by Swarnendu0123** [comment]
Attach video!

**Comment by anisharma07** [comment]
> Attach video!

https://github.com/oxiton-foundation/click-metrics/assets/126901961/613ef1c5-06a7-45d7-81af-98abc6ded8eb

Created new folder types in src directory ...

**Comment by Swarnendu0123** [comment]
Thanks!

**Pull request merged**